app/pages/index.tsx

```diff
@@ -5,11 +5,30 @@ import { useAuth } from '../hooks/useAuth'
 import { useRouter } from 'next/router'
 import { Header } from '../components/Header'
 import { UserMenu } from '../components/UserMenu'
+import type * as grpcWeb from 'grpc-web'
+import { WardServiceClient } from '../generated/Ward-svcServiceClientPb'
+import type { GetWardResponse } from '../generated/ward-svc_pb'
+import { GetWardRequest } from '../generated/ward-svc_pb'
 
 const Home: NextPage = () => {
   const router = useRouter()
   const { user, logout, accessToken } = useAuth(() => router.push({ pathname: '/login', query: { back: true } }))
 
+  const getWard = () => {
+    const wardService = new WardServiceClient('https://staging-api.helpwave.de/task-svc', null, null)
+
+    const req = new GetWardRequest()
+    req.setId('11402cd5-4748-43b0-9868-cc0dd71244b3')
+
+    wardService.getWard(req, null, (err: grpcWeb.RpcError, res: GetWardResponse) => {
+      if (err) {
+        console.error('call failed', err)
+        return
+      }
+      console.log('call successful', res.getId(), res.getName())
+    })
+  }
```

Comment:
can you integrate this into the existing fetch+typechecking+mocking code? (namely [this](https://github.com/helpwave/web/blob/main/app/utils/api.ts); it should behave similar to [this here](https://github.com/helpwave/web/blob/main/app/utils/login.ts#L49-L57))

Reply:
if that is not possible at least integration with `react-query` or `swr`
(the api of the generated code looks atrocious; some kind of nice wrapper around it is needed; what this wrapper will look like is up for debate / depends on how well the generated code can be encapsulated)

Reply:
It is only an example. Beside of the callback and error handling, IMO the API its not that _atrocious_. If we want to have a _"fetch-typechecking+mocking"_ wrapper (and yes, we should), we need to handle [gRPC Server-Side-Streaming](https://github.com/grpc/grpc-web#server-side-streaming) as well.

How would you implement such a wrapper? And should the implementation take place in the scope of this PR?

Reply:
I've adopted to promises instead of the callbacks. We should merge this example and move on as explained above.

Reply:
#88